### PR TITLE
#163388143 authenticated users should be able to favorite articles

### PR DESCRIPTION
**Pull request by asirvex**
#### What does this PR do?
Creates a feature that enables a user to favorite and unfavorite an article

#### Description of Task to be completed?
- create a route for favoriting feature
- write tests for the feature

#### How should this be manually tested?
1. Clone the repo

```
$ git clone https://github.com/andela/ah-the-phoenix.git
```
1. Change it into the project directory

```
$ cd ah-the-phoenix
```
Checkout to the Feature branch

```
$ git checkout ft-favorite-article-163388143
```
Run the server
   `python manage.py runserver`

1.  Favorite an article
      Send a `PUT` request to  `http://127.0.0.1:8000/api/v1/articles/<slug>/favorite/`
favoriting

2. Get favorited articles
      Send a `GET` request to  `http://127.0.0.1:8000/api/v1/favorites/`
get fav articles

3. Unfavorite an article
      Send a `DELETE` request to  `http://127.0.0.1:8000/api/v1/articles/<slug>/favorite/`
unfavoriting

#### Any background context you want to provide?
None

#### What are the relevant pivotal tracker stories?

 [#163388143](https://www.pivotaltracker.com/story/show/163388143)

### CODE REVIEW

File: authors/apps/articles/models.py [hunk 1]
```diff
@@ -57,3 +57,9 @@ class Rating(models.Model):
 
     def __str__(self):
         return self.user_rating
+
+
+class Favorite(models.Model):
+    """Contains the user and article favorited."""
+    article = models.ForeignKey(Article, on_delete=models.CASCADE)
+    user = models.ForeignKey(User, on_delete=models.CASCADE)
```
Comment (sirghiny) [inline authors/apps/articles/models.py]: Include a `related_name` for querying from the other side.

File: authors/apps/authentication/tests/base_test.py [hunk 2]
```diff
@@ -320,6 +321,16 @@ def rate_article_url(self):
         rate_article_url = reverse("articles:rating", args=["rate-this"])
         return rate_article_url
 
+    def favorite_article_url(self):
+        token = self.authenticate_user(self.auth_user2_data).data["token"]
+        self.client.post(self.articles_url,
+                         self.rating_article,
+                         format='json',
+                         HTTP_AUTHORIZATION=f'token {token}')
+        rate_article_url = reverse(
```
Comment (sirghiny) [inline authors/apps/authentication/tests/base_test.py:342]: Shouldn't the variable name be `favourite...`

File: authors/apps/articles/tests/test_favorites.py [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+        self.assertEqual(response.data,
+                         {"error": "you have already favorited this article"})
+
+    def test_unsuccessful_unfavorite(self):
+        token = self.authenticate_user(self.auth_user_data).data["token"]
+        response = self.client.delete(
+            self.favorite_article_url(), HTTP_AUTHORIZATION=f'token {token}'
+        )
+        self.assertEqual(
```
Comment (dev-jey) [inline authors/apps/articles/tests/test_favorites.py:37]: If possible, you could assert for the status code first, before asserting for messages

Comment (asirvex) [reply authors/apps/articles/tests/test_favorites.py:37]: I've added the line

**Pull request merged**